Automattic/jetpack
# Jetpack Sync: Remove 'admin_action_update' hook from Sync Plugins module

The `admin_action_update` hook is no longer fired when making plugin edits in the `plugin-editor` screen, and as of WP 4.9 we added the `wp_ajax_edit-theme-plugin-file` listener (which covers detecting plugin edits in the `plugin-editor` screen) via https://github.com/Automattic/jetpack/pull/8459 anyways.
This PR removes the `admin_action_update` hook from the Plugins module and its corresponding action.

## Proposed changes:

* `Automattic\Jetpack\Sync\Modules\Plugins`: Remove `admin_action_update` hook and `check_plugin_edit` method.

### Other information:


## Jetpack product discussion


p9dueE-8ca-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Code Review: Any WP version we support (6.4 or higher) where the `admin_action_update` hook is fired in `plugin-editor` screen (when making plugin edits) while `wp_ajax_edit-theme-plugin-file` is not? 
* Edit a plugin file in the Plugin Editor and confirm the corresponding action is synced.
